### PR TITLE
First draft of node migration article.

### PR DESCRIPTION
**Pull request by amouat**
## Type of change
New article on migrating Node.js apps.

### CODE REVIEW

**Comment by amouat** [comment]
I haven't mentioned much about npm being removed from latest in this article. Should I write it from the PoV that the change has already happened?

File: content/chainguard/migration/migrating-node.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+EXPOSE 3000
+
+CMD npm start
+```
+
+Directly porting to Chainguard Images with the least number of changes results in this Dockerfile:
+
+```Docker
+FROM cgr.dev/chainguard/node:latest-dev
```
Comment (smythp) [inline content/chainguard/migration/migrating-node.md:136]: I guess a part of me wants a link to an pp and package.json file, even if they're extremely minimal, just something to get these to run without errors. Adding them here does seem like a lot of clutter, though. In mine I linked to a repo that has things set up to work with needed app and requirements . I don't know, maybe it's overkill.

File: content/chainguard/migration/migrating-node.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+title: "Migrating to Node.js Chainguard Images"
+linktitle: "Node Migration"
+type: "article"
+description: "Guidance on how to migrate Node.js Dockerfile workloads to use Chainguard Images"
+date: 2024-05-03T15:56:52-07:00
+lastmod: 2024-05-03T15:56:52-07:00
+draft: false
+tags: ["Images", "Product", "Conceptual"]
+images: []
+weight: 700
```
Comment (smythp) [inline content/chainguard/migration/migrating-node.md:11]: Not sure what order we want these in, but we probably want something explicit. I put mine as 705, the PHP one is 700. Maybe 703 or 710?

**Review by smythp — COMMENTED** [review]
My recommendations are mostly fiddly stuff. I guess my biggest suggestion is to have some kind of example app.js / package.json so that things work if folks are just running through this. A good guide

**Review by smythp — APPROVED** [review]

File: content/chainguard/migration/migrating-node.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+## Chainguard Node.js Images
+
+The Node.js image comes in two main flavours; runtime images intended for production usage and
+builder images intended for use in the build-step of multi-stage builds. The builder images are
+distinguished by the `-dev` suffix (e.g., `latest-dev`).
+
+The production images are intentionally as minimal as possible. They have enough dependencies to run
+a Node.js application but no more. There is no package manager or shell, which reduces the attack
+surface of the image, but also means it is difficult to extend the image. The builder images have
```
[the remark this answers is not in the document]
Comment (amouat) [reply content/chainguard/migration/migrating-node.md:33]: I think it's best to be honest; this is why we have the -dev images. The hard to extend is sort of intentional.

File: content/chainguard/migration/migrating-node.md [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+more build tooling as well as a shell and package manager, allowing them to be easily extended. We
+still aim to keep CVE counts as low as possible in the builder images and they can be used in
+production if neccessary (such as when the application requires extra system dependencies at
+runtime), but we recommend using the builder image as the first step in a mult-stage build with the
+production image as the base for the final image.
+
+This extremely minimal approach to the runtime image is sometimes known as "distroless". For a
+deeper exploration of distroless images and their differences from standard base images, refer to
+the guide on [Getting Started with Distroless
+images](/chainguard/chainguard-images/getting-started-distroless/).
```
[the remark this answers is not in the document]
Comment (amouat) [reply content/chainguard/migration/migrating-node.md:43]: Yeah, I saw what you did with the short codes. Let me look at that...

File: content/chainguard/migration/migrating-node.md (hunk 1 above)
Comment (amouat) [reply content/chainguard/migration/migrating-node.md:136]: I think it's a fair point. I was considering it but was having problems finding a minimal example.

Comment (amouat) [reply content/chainguard/migration/migrating-node.md:155]: I've added this (https://github.com/chainguard-dev/cg-images-node-migration)

**Review by ltagliaferri — APPROVED** [review]

**Pull request merged**